resolwe_bio/processes/fusion_calling/arriba.py

```diff
@@ -0,0 +1,132 @@
+"""Gene fusion detection with Arriba."""
+
+from pathlib import Path
+from plumbum import TEE
+from resolwe.process import (
+    Cmd,
+    DataField,
+    FileField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+class Arriba(Process):
+    """Run Arriba for fusion detection from RNA-Seq data.
+
+    Arriba is a command-line tool for gene fusion detection from RNA-Seq data.
+    This process detects gene fusions from RNA-Seq data using the Arriba tool.
+    The input can be a BAM file from the STAR aligner, and additional optional
+    inputs such as blacklist, protein domains, and known fusions files can be provided.
+    More information about Arriba can be found in the
+    [Arriba manual](https://arriba.readthedocs.io/en/latest/) and in
+    the [original paper](https://genome.cshlp.org/content/31/3/448)
+    The current version of Arriba is 2.4.0.
+    """
+
+    slug = "arriba"
+    name = "Arriba"
+    process_type = "data:genefusions:arriba"
+    version = "1.0.0"
+    category = "RNA-Seq"
+    scheduling_class = SchedulingClass.BATCH
+    entity = {"type": "sample"}
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/rnaseq:6.2.0"}
+        },
+        "resources": {
+            "cores": 4,
+            "memory": 16384,
+        },
+    }
+    data_name = '{{ bam|name|default("?") }}'
+
+    class Input:
+        """Input fields for Arriba process."""
+
+        bam = DataField("alignment:bam:star", label="Input BAM file from STAR aligner")
+        chim_out_type = StringField(
+            label="Chimeric output type",
+            description="Specify the chimeric output type.",
+            choices=["WithinBam", "SAMSeparateOld"],
+            default="WithinBam",
+        )
+        output_file = StringField(
+            label="Output file",
+            description="Name of the output file for predicted fusions.",
+            default="fusions.tsv",
+        )
+        discarded_fusions_file = StringField(
+            label="Discarded fusions file",
+            description="Name of the output file for discarded fusions.",
+            default="discarded_fusions.tsv",
+        )
+        gtf_file = DataField(
+            "annotation",
+            label="GTF file",
+            description="Annotation file in GTF format.",
+        )
+        blacklist_file = DataField(
+            "genelist",
```

Comment:
This should be an existing Data type. If a dedicated "upload" process that creates such type is not available or is not be introduced, a [general process](https://github.com/genialis/resolwe/blob/master/resolwe/toolkit/processes/files.py) to upload "files" should be used. Be advised that when using this process, there is no built-in file validation, and the uploaded file is stored "as is". The same applies bellow.

Reply:
Hmm, blacklists are simple .tsv files with listed genomic coordinates (e.g. 14:1234-4321) that should be ignored by Arriba. Since these files are relatively big, the idea is to store them in an S3 bucket and put them into a collection in Expressions. Thus I don't think that anything will be actually uploaded at this point.